src/main/puzzle/playfield.gd

```diff
@@ -5,20 +5,20 @@ Stores information about the game playfield: writing pieces to the playfield, ca
 or whether a box was made, pausing and playing sound effects
 """
 
-# signal emitted before a line is cleared
-signal before_line_cleared(y, total_lines, remaining_lines)
-
-# signal emitted when a line is cleared
-signal line_cleared(y, total_lines, remaining_lines)
-
-# signal emitted when a box (3x3, 3x4, 3x5) is made
+signal before_line_cleared(y, total_lines, remaining_lines, box_ints)
+signal line_cleared(y, total_lines, remaining_lines, box_ints)
 signal box_made(x, y, width, height, color_int)
+signal piece_written
```

Comment:
Document this `piece_written` signal, it's a little surprising.

Reply:
Renamed to `after_piece_written` to be less surprising. Also rearranged the signals in the order they're fired.